test/integration/link_opts.t/run.t

```diff
@@ -1,50 +1,55 @@
   $ ocamlc -bin-annot test.ml
   $ mkdir h
-  $ odoc compile --output-dir h --parent-id pkg/doc page.mld
-  $ odoc compile --output-dir h --parent-id pkg/lib/libname test.cmt
+  $ odoc compile --output-dir h --parent-id pkg page.mld
+  $ odoc compile --output-dir h --parent-id pkg/libname test.cmt
 
-No library or package are passed, no error. This ensures compatibility with Odoc 2.
+No -P or -L passed, no error. This ensures compatibility with Odoc 2.
 
-  $ odoc link -P pkg:h/pkg/doc h/pkg/lib/libname/test.odoc
-  $ odoc link -P pkg:h/pkg/doc h/pkg/doc/page-page.odoc
-  $ odoc link -L libname:h/pkg/lib/libname h/pkg/lib/libname/test.odoc
-  $ odoc link -L libname:h/pkg/lib/libname h/pkg/doc/page-page.odoc
+  $ odoc link h/pkg/libname/test.odoc
+  $ odoc link h/pkg/page-page.odoc
+  $ odoc link h/pkg/libname/test.odoc
+  $ odoc link h/pkg/page-page.odoc
 
-Current library is not passed:
+A package can be passed, either with `--current-package` or by being below a
+`-P`.
 
-  $ odoc link -P pkg:h/pkg/doc -L otherlib:h/otherpkg h/pkg/lib/libname/test.odoc
-  ERROR: The input file must be part of a directory passed as -L
-  [1]
-  $ odoc link -P pkg:h/pkg/doc -L otherlib:h/otherpkg h/pkg/doc/page-page.odoc
+ For modules:
 
-Current package is not passed:
+  $ odoc link --current-package pkg -P pkg:h/pkg h/pkg/libname/test.odoc
+  $ odoc link -P pkg:h/pkg h/pkg/libname/test.odoc
 
-  $ odoc link -P otherpkg:h/otherpkg/doc -L libname:h/pkg/lib/libname h/pkg/lib/libname/test.odoc
-  $ odoc link -P otherpkg:h/otherpkg/doc -L libname:h/pkg/lib/libname h/pkg/doc/page-page.odoc
-  ERROR: The input file must be part of a directory passed as -P
-  [1]
+ For pages:
+
+  $ odoc link --current-package pkg -P pkg:h/pkg2 h/pkg/page-page.odoc
+  $ odoc link -P pkg:h/pkg h/pkg/page-page.odoc
+
+It is not required to be below a `-L`, even for modules:
 
-Specified current package is wrong:
+  $ odoc link -P pkg:h/pkg -L otherlib:h/otherpkg h/pkg/libname/test.odoc
```

Comment:
nit: do you want `-L otherlib:h/otherpkg/otherlib` here instead?

Reply:
Yes. I've cleaned up the test a bit (some options were not really necessary for the test and made it less readable. Hopefully it is better now!